app.rb

```diff
@@ -0,0 +1,157 @@
+require './book'
+require './teacher'
+require './student'
+require './rental'
+
+class App
+  attr_reader :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # list all books
+  def list_books
+    if @books.length.zero?
+      puts 'There are no books at the moment.'
+    else
+      @books.each_with_index { |book, i| puts "#{i}) Title: \"#{book.title}\", Author: #{book.author}" }
+    end
+  end
+
+  # list all people
+
+  def list_people
+    if @people.length.zero?
+      puts 'There are no people at the moment.'
+    else
+      @people.each_with_index do |person, i|
+        puts "#{i}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  # create a person[teacher or student]
+  def add_student(student)
+    @people << student
+  end
+
+  def add_teacher(teacher)
+    @people << teacher
+  end
+
+  def create_person_type(type, age, name)
+    case type
+    when 1
+      permission = ''
+      until %w[Y N].include?(permission)
+        puts 'Has parent permission? [Y/N]:'
+        permission = gets.chomp.capitalize
+      end
+      student = Student.new(nil, age, name, parent_permission: permission == 'Y')
+      add_student(student)
+    when 2
+      puts 'Specialization:'
+      specialization = gets.chomp.capitalize
+      teacher = Teacher.new(specialization, age, name)
+      add_teacher(teacher)
+    end
+  end
+
+  def create_person
+    person_type = 0
+    until [1, 2].include?(person_type)
+      puts 'Do you want to create a student (1) or do you want to create a teacher (2)? [Input the number]:'
+      person_type = gets.chomp.to_i
+    end
+
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp
+    create_person_type(person_type, age, name)
+    person = person_type == 1 ? 'Student' : 'Teacher'
+    puts "#{person} created successfully."
+  end
+
+  # create a book
+
+  def add_book(book)
+    @books << book
+  end
+
+  def create_book
+    title = ''
+    author = ''
+    while title.empty?
+      print 'Title: '
+      title = gets.chomp
+    end
+    while author.empty?
+      puts 'Author: '
+      author = gets.chomp
+    end
+    book = Book.new(title, author)
+    add_book(book)
+    puts 'Book created successfully.'
+  end
+
+  # create a rental
+  def add_rental(rental)
+    @rental << rental
+  end
+
+  def create_rental
+    i_book = nil
+    i_person = nil
+    until (0..@book.length - 1).include? i_book
```

Comment:
- [ ] It would be better if our app follows the UI design. However, there are some errors preventing us from achieving that. Here is the screenshot of the error so that we can investigate and fix it accordingly
![Screenshot from 2022-10-08 00-38-47](https://user-images.githubusercontent.com/46388113/194678927-0d6d791b-31dc-41a2-8b7e-59d59f9705d3.png)

Trying to create a rental (5) resulted in the error above